src/sqlfluff/dialects/dialect_ansi.py

```diff
@@ -526,13 +528,19 @@
     ),
     # This is a placeholder for other dialects.
     SimpleArrayTypeGrammar=Nothing(),
+    # Base Expression element is the right thing to reference for everything
+    # which functions as an expression, but could include literals.
     BaseExpressionElementGrammar=OneOf(
         Ref("LiteralGrammar"),
         Ref("BareFunctionSegment"),
         Ref("IntervalExpressionSegment"),
         Ref("FunctionSegment"),
         Ref("ColumnReferenceSegment"),
         Ref("ExpressionSegment"),
+        Sequence(
+            Ref("DatatypeSegment"),
+            Ref("LiteralGrammar"),
+        ),
```

Comment:
Unfortunately not. The bare `LiteralGrammar` needs to go before `ExpressionSegment` otherwise it just wraps all the literals in expressions, and the `DatatypeSegment` version needs to go after because otherwise it thinks everything is a data type.